crystal-map-connector/src/main/kotlin/ch/ergon/dope/extension/type/arithmetic/NumberInfix.kt

```diff
@@ -41,6 +48,9 @@ fun Number.mul(number: CMField<out Number>) = toDopeType().mul(number.toDopeType
 
 fun CMField<out Number>.mul(number: Number) = toDopeType().mul(number.toDopeType())
 
+fun <KotlinType : Any, MapType : Number> KotlinType.mul(other: CMConverterField<KotlinType, MapType>) =
+    other.typeConverter.write(this)!!.toDopeType().mul(other.toDopeType())
```

Comment:
We shouldn't use !! in our code, just because if it returns _null_ (which is obviously allowed) our program crashes. Since _other_ is also of type _CMField_, can't you treat it lika a _CMField_ and call the corresponding _toDopeType_ method (if the typeconverter returns null)?

Reply:
That we shouldn't use the !! I completly agree, but I don't think just calling the `toDopeQuery` for the CMField will solve this problem, because we want `this` the `KotlinType` to be converted and not the  `other`.
I don't have a better solution right now. Maybe we should create a ticket for making this cleaner?

Reply:
use a _requireNotNull_

Reply:
I added a function to do that